### PR TITLE
Added Group `bitswap`

### PR DESCRIPTION
**Pull request by RichardLitt**
I added `ipfs bitswap stat`, `wantlist`, and `unwant`. I also added an Error object data structure for better error logging for future requests.

### CODE REVIEW

**Comment by RichardLitt** [comment]
cc @dignifiedquire @diasdavid

File: apiary.apib [hunk 1]
```diff
@@ -398,11 +398,318 @@ MerkleDAG.
...
+        Transfer-Encoding: chunked
+        ```
+
+    + Body
+
+        ```
+        ```
+
+## wantlist [GET /bitswap/wantlist{?peer}]
+Show blocks currently on the wantlist.
```
Comment (daviddias) [inline apiary.apib:632]: - of a peer or itself

File: apiary.apib [hunk 2]
```diff
@@ -398,11 +398,318 @@ MerkleDAG.
...
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Content-Type: application/json
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Fri, 29 Jan 2016 21:09:22 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (Error)
+        - Message: 'multihash too short. must be \u003e 3 bytes'
```
Comment (daviddias) [inline apiary.apib:709]: this is a weird error - `\u003e 3 bytes`

File: apiary.apib [hunk 3]
```diff
@@ -398,11 +398,318 @@ MerkleDAG.
...
+        ```
+        Access-Control-Allow-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Ouptut, X-Chunked-Output
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        Date: Thu, 21 Jan 2016 17:09:00 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Body
```
Comment (diasdavid) [inline apiary.apib]: stat doesn't return any body?

File: apiary.apib [hunk 4]
```diff
@@ -398,11 +398,318 @@ MerkleDAG.
...
 
-## unwant
+        ```
+        {
+          "ProvideBufLen": 0,
+          "Wantlist": null,
+          "Peers": [
```
Comment (daviddias) [inline apiary.apib:448]: Are these the peers that we have want lists? Or peers connected in general?

File: apiary.apib [hunk 5]
```diff
@@ -398,11 +398,318 @@ MerkleDAG.
...
+
+    + Attributes (Error)
+        - Message: 'incorrectly formatted key: '
+        - Code: 0
+
+    + Body
+
+        ```
+        {
+          "Message": "incorrectly formatted key: ",
```
Comment (daviddias) [inline apiary.apib]: Inconsistent error messages. https://github.com/ipfs/http-api-spec/pull/24/files#r51342821 is the same issue, but shows a different error message.

**Comment by daviddias** [comment]
Looks like we are starting to get some material we can open issues on go-ipfs :)

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib:632]: I don't understand the context for this comment: can you rephrase?

File: apiary.apib (hunk 2 above)
Comment (RichardLitt) [inline apiary.apib:709]: It's the greater than sign, which is escaped because the content is application/json.

File: apiary.apib (hunk 4 above)
Comment (RichardLitt) [inline apiary.apib:448]: I don't know. @whyrusleeping ?

Comment (whyrusleeping) [inline apiary.apib:448]: @diasdavid right now those are the same thing

File: apiary.apib (hunk 1 above)
Comment (daviddias) [inline apiary.apib:632]: because if the param 'peer' is passed, we will see the want list that we have from that peer, if we don't pass anything, our own want list will be shown

File: apiary.apib (hunk 2 above)
Comment (daviddias) [inline apiary.apib:709]: :+1:

**Comment by daviddias** [comment]
Seems like other than the tiny tid bit here https://github.com/ipfs/http-api-spec/pull/24#discussion_r53509224, this is ready to be rebased and clean merged?

**Comment by dignifiedquire** [comment]
needs rebase

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib:632]: Covered in the options. See updated code.

File: apiary.apib (hunk 5 above)
Comment (RichardLitt) [inline apiary.apib]: Not the same error message, unless blocks can only be added if they are Multihashes. @whyrusleeping must the `key` always be a multihash? If so, then there is a bug in [here](https://github.com/ipfs/go-ipfs/blob/master/core/commands/bitswap.go#L57-L60) as it throws an error for empty strings.

**Comment by RichardLitt** [comment]
Found a bug in the docs where the ability to add multiple keys isn't mentioned. Open PR. https://github.com/ipfs/go-ipfs/pull/2409

@whyrusleeping must the key in `ipfs bitswap unwant <key>` always be a multihash?  If so, then there is a bug in [here](https://github.com/ipfs/go-ipfs/blob/master/core/commands/bitswap.go#L57-L60) as it throws an error for empty strings.

**Comment by RichardLitt** [comment]
Fixed that bug, #2409 is closed. Key is synonymous with multihash, so, yes, and b58 always returns an empty string when decoded like that. So. This is all good.
